### PR TITLE
Export some helpers and add some types

### PR DESCRIPTION
**Pull request by hsanjuan**
The purpose of this rather superficial changes is to better support custom, ad-hoc tooling that re-uses logic present in sentinel-visor.

### CODE REVIEW

File: commands/setup.go [hunk 1]
```diff
@@ -34,13 +34,16 @@ import (
...
-	db     *storage.Database
+	Opener lens.APIOpener
+	Closer lens.APICloser
+	DB     *storage.Database
 }
 
-func setupStorageAndAPI(cctx *cli.Context) (context.Context, *RunContext, error) {
+// SetupStorageAndAPI setups of the sentinel database and returns a
+// ready-to-use RunContext with Openers and Closers using the chosen lens.
+func SetupStorageAndAPI(cctx *cli.Context) (context.Context, *RunContext, error) {
```
Comment (hsanjuan) [inline commands/setup.go:46]: Purpose is that we can call SetupStorageAndAPI from a different program easily and then be able to handle DB closing from there.

File: model/actors/common/actors.go [hunk 2]
```diff
@@ -28,13 +31,33 @@ func (a *Actor) PersistWithTx(ctx context.Context, tx *pg.Tx) error {
...
+	if len(actors) == 0 {
+		return nil
+	}
+	if _, err := tx.ModelContext(ctx, &actors).
+		OnConflict("do nothing").
+		Insert(); err != nil {
+		return fmt.Errorf("persisting actors: %w", err)
+	}
+	return nil
+}
```
Comment (hsanjuan) [inline model/actors/common/actors.go:51]: Doing this should be much faster than doing one insert per actor.

File: model/actors/miner/sectorevents.go [hunk 3]
```diff
@@ -42,6 +42,11 @@ type MinerSectorEventList []*MinerSectorEvent
 func (l MinerSectorEventList) PersistWithTx(ctx context.Context, tx *pg.Tx) error {
 	ctx, span := global.Tracer("").Start(ctx, "MinerSectorEventList.PersistWithTx", trace.WithAttributes(label.Int("count", len(l))))
 	defer span.End()
+
+	if len(l) == 0 {
+		return nil
+	}
```
Comment (hsanjuan) [inline model/actors/miner/sectorevents.go:48]: psql errors when being given a nil slice.

File: model/actors/miner/task.go [hunk 4]
```diff
@@ -82,10 +84,74 @@ func (res *MinerTaskResult) Persist(ctx context.Context, db *pg.DB) error {
...
+type MinerTaskLists struct {
+	MinerInfoModel           MinerInfoList
+	FeeDebtModel             MinerFeeDebtList
+	LockedFundsModel         MinerLockedFundsList
+	CurrentDeadlineInfoModel MinerCurrentDeadlineInfoList
+	PreCommitsModel          MinerPreCommitInfoList
+	SectorsModel             MinerSectorInfoList
+	SectorEventsModel        MinerSectorEventList
+	SectorDealsModel         MinerSectorDealList
+}
```
Comment (hsanjuan) [inline model/actors/miner/task.go:108]: Allows processing all miners in a tipset, collecting all the models and then persisting that. Should help towards making a single Tx per tipset and be faster than using MinerTaskResultList.

**Review by frrist — APPROVED** [review]
Good stuff!

**Pull request merged**